pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java

```diff
@@ -1510,8 +1516,15 @@ public void uploadFunction(final InputStream uploadedInputStream, final String p
         }
     }
 
-    public StreamingOutput downloadFunction(final String path) {
+    public StreamingOutput downloadFunction(final String path, String clientRole) {
 
+        if (worker().getWorkerConfig().isAuthorizationEnabled()) {
+            // skip authorization if client role is super-user
+            if (!isSuperUser(clientRole)) {
```

Comment:
@rdhabalia this is not correct.  This means that only super users will be able to download jars.  Currently, for the kubernetes runtime, we are relying on this API to download the JARs into containers.  We shouldn't restrict this to only super users, we should come up of a way for users to have ACLs to what they can or cannot download.

Reply:
ok.. this PR was created to address V2 api where authentication is failing for v2 api because it was passing null as clientId. for this particular usecase, let me remove the check for now.

Reply:
done